### PR TITLE
[Rebalance] Add unit tests for some edge cases

### PR DESCRIPTION
**Pull request by maxfirmbach**
## Description and Context

This PR adds some unit testing for edge cases that might happen during graph rebalancing, e.g.:
- Serial case: input graph should be equal to output graph (we should not guard with myrank == 0)
- One processor holds the full graph with others beeing empty: after rebalancing the empty procs should be populated
- Empty graph: should throw before rebalancing
- Empty discretization: should throw before building a monolithic graph

## Related Issues and Pull Requests

Part of #594

### CODE REVIEW

**Comment by maxfirmbach** [comment]
This is just an initial effort to have unit testing for the rebalancing at all. Other cases will be added in the near future.

**Review by amgebauer — APPROVED** [review]

File: src/core/rebalance/tests/4C_rebalance_graph_test.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    graph.optimize_storage();
+
+    Teuchos::ParameterList rebalance_params;
+    rebalance_params.set("partitioning method", "hypergraph");
+
+    auto rebalanced_graph = Core::Rebalance::rebalance_graph(graph, rebalance_params);
+
+    EXPECT_EQ(rebalanced_graph->num_global_nonzeros(), graph.num_global_nonzeros());
+    EXPECT_EQ(rebalanced_graph->num_local_nonzeros(), graph.num_local_nonzeros());
+    EXPECT_EQ(rebalanced_graph->num_global_nonzeros(), rebalanced_graph->num_local_nonzeros());
```
Comment (mayrmt) [inline src/core/rebalance/tests/4C_rebalance_graph_test.cpp:75]: Can't' you do a `same_as()` query to compare both graphs?

Comment (maxfirmbach) [reply src/core/rebalance/tests/4C_rebalance_graph_test.cpp:75]: Does that exist for graphs? I have never seen this method.

Comment (mayrmt) [reply src/core/rebalance/tests/4C_rebalance_graph_test.cpp:75]: I thought that it exists. But you are right: it does not. Sorry.
